### ISSUE TITLE
Pip module doesn't handle "develop" packages

### ISSUE THREAD

**Issue by nigelsim** [saltstack/salt#30799]
When I tried to run a pip.installed state on a server that had some manually installed packaged I got the following error:

```
     ID: pywin32
    Function: pip.installed
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "C:\salt\bin\lib\site-packages\salt\state.py", line 1594, in call
                  **cdata['kwargs'])
                File "C:\salt\bin\lib\site-packages\salt\loader.py", line 1491, in wrapper
                  return f(*args, **kwargs)
                File "C:\salt\bin\lib\site-packages\salt\states\pip_state.py", line 621, in installed
                  upgrade, user, cwd, bin_env)
                File "C:\salt\bin\lib\site-packages\salt\states\pip_state.py", line 179, in _check_if_installed
                  user=user, cwd=cwd)
                File "C:\salt\bin\lib\site-packages\salt\modules\pip.py", line 953, in list_
                  version_, name = line.split('#egg=')
              ValueError: need more than 1 value to unpack
     Started: 10:27:00.536000
    Duration: 569.0 ms
     Changes:   
```

The error comes in parsing the output of `pip freeze`. This is the output that we get on that machine

```
boto==2.19.0
crontab==0.20.2
pytz==2014.10
pywin32==218
PyYAML==3.11
requests==2.1.0
stomp.py==4.1.9
virtualenv==13.1.2
-e vsaeclient==0.1.dev0
```

The last package is an internal python package that we installed from source in "edit" mode.

**Comment by jfindlay** [comment]
@nigelsim, thanks for the report.  What version of salt are you using?  Also, will you post the state that is failing?

**Comment by nigelsim** [comment]
This was the state that did it, but I actually can't reproduce this error anymore. After I removed the offending library I tried to get a similarly broken install in a sandbox, but everything I've tried produces a working install. My best guess is that this particular install, on Windows, has been upgraded, and upgraded, and somewhere in its history a bad develop link has been put in by an old version of python/setuptools. 

I think this issue should probably be closed, but hopefully it will give a hint to someone else hitting this error in the future.

```
pywin32:
  pip.installed:
    - name: pypiwin32
    - upgrade: True
    - bin_env: "c:\python27\scripts\pip.exe"
```
